# Fundamentals of Data Analysis_Assessment 2019

### Title and explaination
- Ref 
- 1. https://mc.ai/studying-suicide-rates-from-1985-to-2016-prediction-using-machine-learning/
- 2. https://www.geeksforgeeks.org
- 3. https://pbpython.com/pandas_dtypes.html (Clean data)
- 

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
df=pd.read_csv("suicide_rates_overview_1985_to_2016.csv")
#!pwd_needed this to confirm my file path to load the dataset

In [2]:
df.head()

country  year     sex          age  suicides_no  population  \
0  Albania  1987    male  15-24 years           21      312900   
1  Albania  1987    male  35-54 years           16      308000   
2  Albania  1987  female  15-24 years           14      289700   
3  Albania  1987    male    75+ years            1       21800   
4  Albania  1987    male  25-34 years            9      274300   

   suicides/100k pop country-year  HDI for year  gdp_for_year ($)   \
0               6.71  Albania1987           NaN      2,156,624,900   
1               5.19  Albania1987           NaN      2,156,624,900   
2               4.83  Albania1987           NaN      2,156,624,900   
3               4.59  Albania1987           NaN      2,156,624,900   
4               3.28  Albania1987           NaN      2,156,624,900   

   gdp_per_capita ($)       generation  
0                 796     Generation X  
1                 796           Silent  
2                 796     Generation X  
3                 796  G.I. Generation  
4                 796          Boomers

In [3]:
df.sort_values(by= ['country'])

country  year     sex          age  suicides_no  population  \
0         Albania  1987    male  15-24 years           21      312900   
167       Albania  2002    male   5-14 years            1      319473   
168       Albania  2003  female    75+ years            6       49088   
169       Albania  2003    male  55-74 years           16      201520   
170       Albania  2003    male  35-54 years           28      386196   
...           ...   ...     ...          ...          ...         ...   
27649  Uzbekistan  1997  female  35-54 years           74     2011500   
27650  Uzbekistan  1997    male   5-14 years           37     3159000   
27651  Uzbekistan  1997  female   5-14 years            9     3071700   
27637  Uzbekistan  1996  female  35-54 years           74     1906900   
27819  Uzbekistan  2014  female  55-74 years           21     1438935   

       suicides/100k pop    country-year  HDI for year  gdp_for_year ($)   \
0                   6.71     Albania1987           NaN      2,156,624,900   
167                 0.31     Albania2002           NaN      4,435,078,648   
168                12.22     Albania2003           NaN      5,746,945,913   
169                 7.94     Albania2003           NaN      5,746,945,913   
170                 7.25     Albania2003           NaN      5,746,945,913   
...                  ...             ...           ...                ...   
27649               3.68  Uzbekistan1997           NaN     14,744,603,774   
27650               1.17  Uzbekistan1997           NaN     14,744,603,774   
27651               0.29  Uzbekistan1997           NaN     14,744,603,774   
27637               3.88  Uzbekistan1996           NaN     13,948,892,216   
27819               1.46  Uzbekistan2014         0.675     63,067,077,179   

       gdp_per_capita ($)    generation  
0                     796  Generation X  
167                  1573    Millenials  
168                  2021        Silent  
169                  2021        Silent  
170                  2021       Boomers  
...                   ...           ...  
27649                 724       Boomers  
27650                 724    Millenials  
27651                 724    Millenials  
27637                 703       Boomers  
27819                2309       Boomers  

[27820 rows x 12 columns]

In [4]:
print (df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 27820 entries, 0 to 27819
Data columns (total 12 columns):
country               27820 non-null object
year                  27820 non-null int64
sex                   27820 non-null object
age                   27820 non-null object
suicides_no           27820 non-null int64
population            27820 non-null int64
suicides/100k pop     27820 non-null float64
country-year          27820 non-null object
HDI for year          8364 non-null float64
 gdp_for_year ($)     27820 non-null object
gdp_per_capita ($)    27820 non-null int64
generation            27820 non-null object
dtypes: float64(2), int64(4), object(6)
memory usage: 2.5+ MB
None


In [5]:
print (("The following are the uniques values for Generation"), df.generation.unique())

The following are the uniques values for Generation ['Generation X' 'Silent' 'G.I. Generation' 'Boomers' 'Millenials'
 'Generation Z']


In [6]:
print (("The following are the unique values for the years:"), df.year.unique())
df['year'].dtype

The following are the unique values for the years: [1987 1988 1989 1992 1993 1994 1995 1996 1997 1998 1999 2000 2001 2002
 2003 2004 2005 2006 2007 2008 2009 2010 1985 1986 1990 1991 2012 2013
 2014 2015 2011 2016]


dtype('int64')

In [7]:
df.describe()

year   suicides_no    population  suicides/100k pop  \
count  27820.000000  27820.000000  2.782000e+04       27820.000000   
mean    2001.258375    242.574407  1.844794e+06          12.816097   
std        8.469055    902.047917  3.911779e+06          18.961511   
min     1985.000000      0.000000  2.780000e+02           0.000000   
25%     1995.000000      3.000000  9.749850e+04           0.920000   
50%     2002.000000     25.000000  4.301500e+05           5.990000   
75%     2008.000000    131.000000  1.486143e+06          16.620000   
max     2016.000000  22338.000000  4.380521e+07         224.970000   

       HDI for year  gdp_per_capita ($)  
count   8364.000000        27820.000000  
mean       0.776601        16866.464414  
std        0.093367        18887.576472  
min        0.483000          251.000000  
25%        0.713000         3447.000000  
50%        0.779000         9372.000000  
75%        0.855000        24874.000000  
max        0.944000       126352.000000

- Requirement to clean this data! Clearly there are not 27820 countries in the world. From looking at df.head(), it is clear that a number of of indecies are repeated to take the year into account. It is neccessary to count the number of unique entires in Col "Country". The number of countries is compounded by year and age. 

In [8]:
df['country'].nunique()

101

In [18]:
# Examine the number of suicides per country in total
suicides_country = []

for y in df['country'].unique():
    suicides_country.append(sum(df[df['country'] == y]['suicides_no']))
    
#suicides_country

suicides_year = pd.DataFrame(suicides_country, columns=['suicides_country'], index = [df['country'].unique()])
#n_suicides_year['year'] = df['year'].unique()
suicides_year.sort_values(by='suicides_country', ascending = False)
#suicides_year.head()

suicides_country
Russian Federation              1209742
United States                   1034013
Japan                            806902
France                           329127
Ukraine                          319950
...                                 ...
Maldives                             20
Antigua and Barbuda                  11
San Marino                            4
Saint Kitts and Nevis                 0
Dominica                              0

[101 rows x 1 columns]

In [21]:
# Examine the number of suicides per 100k pop in total
suicides_pc = []

for y in df['country'].unique():
    suicides_pc.append(sum(df[df['country'] == y]['suicides/100k pop']))
    
#suicides_country

suicides_pc = pd.DataFrame(suicides_pc, columns=['suicides/100k pop'], index = [df['country'].unique()])
#n_suicides_year['year'] = df['year'].unique()
suicides_pc.sort_values(by='suicides/100k pop', ascending = False)
#suicides_year.head()

suicides/100k pop
Russian Federation              11305.13
Lithuania                       10588.88
Hungary                         10156.07
Kazakhstan                       9519.52
Republic of Korea                9350.45
...                                  ...
Jamaica                           106.44
United Arab Emirates               94.89
Oman                               26.50
Saint Kitts and Nevis               0.00
Dominica                            0.00

[101 rows x 1 columns]

### In this data frame, country
- year from 
- sex: Male or female
- age:
- suicides_no
- population
- suicides/100k pop
- country-year
- HDI for year: The Human Development Index (HDI) is a statistic composite index of life expectancy, education, and per capita income indicators, which are used to rank countries into four tiers of human development. A country scores a higher HDI when the lifespan is higher, the education level is higher, and the gross national income GNI (PPP) per capita is higher
- gdp_for_year (in dollars)
- gdp_per_capita (in dollars)
- generation: Is a steriotypy of different attributes, they include 'Generation X', 'Silent', 'G.I. Generation', 'Boomers', 'Millenials', 'Generation Z'.

In [ ]:
suicides_no_year = []

for y in df['year'].unique():
    suicides_no_year.append(sum(df[df['year'] == y]['suicides_no']))

n_suicides_year = pd.DataFrame(suicides_no_year, columns=['suicides_no_year'])
n_suicides_year['year'] = df['year'].unique()

top_year = n_suicides_year.sort_values('suicides_no_year', ascending=False)['year']
top_suicides = n_suicides_year.sort_values('suicides_no_year', ascending=False)['suicides_no_year']

In [ ]:
n_suicides_year.head()